src/analysis/processing/qgsalgorithmrandomraster.h

```diff
@@ -27,36 +27,238 @@
-    QString svgIconPath() const override { return QgsApplication::iconPath( QStringLiteral( "/algorithms/mAlgorithmRandomRaster.svg" ) ); }
-    QString name() const override;
-    QString group() const override;
-    QString groupId() const override;
-    QString displayName() const override;
-    QStringList tags() const override;
-    QString shortHelpString() const override;
-    QgsRandomRasterAlgorithm *createInstance() const override SIP_FACTORY;
 
   protected:
+
+    /**
+     * Adds specific subclass algorithm parameters. The common parameters, such as raster destination, are automatically
+     * added by the base class.
+     */
+    virtual void addAlgorithmParams() = 0;
+
     bool prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) final;
+
+    /**
+     * Prepares the random number algorithm subclass for execution.
+     */
+    virtual Qgis::DataType getRasterDataType( int typeId ) = 0;
+    virtual bool prepareRandomParameters( const QVariantMap &parameters, QgsProcessingContext &context ) = 0;
+
+    /**
+     * Processes a raster using the generateRandomIntValues method which is implemented in subclasses providing different fuzzy membership types.
+     */
     QVariantMap processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback ) final;
 
+    /**
+     * Virtual methods for random number generation may be overridden by subclassed algorithms to use specific random number distributions.
+     */
+    virtual long generateRandomLongValue( std::mt19937 &mersenneTwister ) = 0;
+    virtual double generateRandomDoubleValue( std::mt19937 &mersenneTwister ) = 0;
+
   private:
     QgsRectangle mExtent;
     QgsCoordinateReferenceSystem mCrs;
     double mPixelSize;
     Qgis::DataType mRasterDataType;
-    int mTypeId;
+};
+
+
+class QgsRandomUniformRasterAlgorithm : public QgsRandomRasterAlgorithmBase
+{
+  public:
+    QgsRandomUniformRasterAlgorithm() = default;
+    QIcon icon() const override { return QgsApplication::getThemeIcon( QStringLiteral( "/algorithms/mAlgorithmRandomRaster.svg" ) ); }
+    QString svgIconPath() const override { return QgsApplication::iconPath( QStringLiteral( "/algorithms/mAlgorithmRandomRaster.svg" ) ); }
+    QString name() const override;
+    QString displayName() const override;
+    QStringList tags() const override;
+    QString shortHelpString() const override;
+    QgsRandomUniformRasterAlgorithm *createInstance() const override SIP_FACTORY;
+
+  protected:
+    void addAlgorithmParams() override;
+    Qgis::DataType getRasterDataType( int typeId ) override;
+    bool prepareRandomParameters( const QVariantMap &parameters, QgsProcessingContext &context ) override;
+    long generateRandomLongValue( std::mt19937 &mersenneTwister ) override;
+    double generateRandomDoubleValue( std::mt19937 &mersenneTwister ) override;
```

Comment:
There may potentially be a small performance boost by swapping "override" for "final" in these classes.